python/opae.admin/opae/admin/tools/opaeuio.py

```diff
@@ -0,0 +1,185 @@
+# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+# INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+# CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+""" Bind/Unbind a PCIe device to/from uio. """
+
+import argparse
+import grp
+import os
+import pwd
+import subprocess
+import sys
+import time
+from opae.admin.dfl import dfl
+
+OPAEUIO_VERSION = '1.0.0'
+
+def parse_args():
+    """Parse script arguments."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument('device', nargs='?',
+                        help='DFL device name')
+    parser.add_argument('-i', '--init', default=False, action='store_true',
+                        help='initialize the given device for uio')
+    parser.add_argument('-r', '--release', default=False, action='store_true',
+                        help='release the given device from uio')
+    parser.add_argument('-d', '--driver', default='dfl-uio-pdev',
+                        help='driver to bind on init')
+    parser.add_argument('-u', '--user', default='root',
+                        help='userid to assign during init')
+    parser.add_argument('-g', '--group', default='root',
+                        help='groupid to assign during init')
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s {}'.format(OPAEUIO_VERSION),
+                        help='display version information and exit')
+    return parser, parser.parse_args()
+
+
+def load_driver(driver):
+    """Load a Linux kernel module via modprobe.
+
+    driver - name of the driver to load.
+    """
+    return subprocess.call(['modprobe', driver])
+
+
+def initialize_uio(device, pac, new_owner, driver):
+    """Bind a DFL device and prepare it for use with uio.
+
+    device - the DFL device name.
+    pac - the dfl object (from dfl.enum()) corresponding to device.
+    new_owner - user:group for the owner of the resulting uio device.
+    driver - the new uio driver to bind.
+
+    Unbind the given DFL device from its current driver, re-assigning it
+    to the driver named by the last parameter. Discover the newly-created
+    uio device, and set the device permissions to rw-rw---- for new_owner,
+    given that new_owner != 'root:root'.
+    """
+    if pac.driver_override.value == driver:
+        print('{} is already bound to {}.'.format(device, driver))
+        return
+
+    print('Unbinding {} from {}.'.format(device, pac.driver.name))
+    pac.unbind()
+
+    print('Setting driver override to {}.'.format(driver))
+    pac.driver_override.value = driver
+
+    print('Loading {}.'.format(driver))
+    load_driver(driver)
+
+    print('Probing {}..'.format(device))
+    with open('/sys/bus/dfl/drivers_probe', 'w') as fp:
```

Comment:
Define as global /sys/bus/dfl/drivers_probe

Reply:
For what benefit?